upb/port/def.inc

```diff
@@ -116,7 +116,7 @@ Error, UINTPTR_MAX is undefined
 #endif
 
 // Macros for function attributes on compilers that support them.
-#ifdef __GNUC__
+#if defined(__GNUC__) || defined(__clang__)
```

Comment:
I am pretty sure that Clang defines `__GNUC__`, so I think this PR is a no-op: https://godbolt.org/z/bcMTaWc57

Reply:
It used to but in recent versions it has stopped. https://clang.llvm.org/docs/UsersManual.html 

```
-fgnuc-version=
This flag controls the value of __GNUC__ and related macros. This flag does not enable or disable any GCC extensions implemented in Clang. Setting the version to zero causes Clang to leave __GNUC__ and other GNU-namespaced macros, such as __GXX_WEAK__, undefined.
```

On a default windows-x64 configuration with clang from MSVC build tools

```
clang version 17.0.3
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\x64\bin
```

```cpp
// clang test.cpp -o test.exe && test

#include <iostream>

int main() {
#ifdef __GNUC__
std::cout << "Gnu!" << std::endl;
#else
std::cout << "Not Gnu!" << std::endl;
#endif

return 0;
}
```

Outputs `Not Gnu!`

Reply:
Running as `clang test.cpp -o test.exe -fgnuc-version=1234 && test` does resolve this and the test program properly prints `Gnu!` however having to add an option specific to make UPB generate optimal code feels rather hacky